iroh/examples/hello-world.rs

```diff
@@ -22,12 +22,17 @@ async fn main() -> anyhow::Result<()> {
     setup_logging();
     // create a new, empty in memory database
     let mut db = iroh::baomap::readonly_mem::Store::default();
+    // create an in-memory doc store (not used in the example)
```

Comment:
Because the `iroh::node::Node` is now generic over the doc store. What we could do is to add a `NoopDocStore` or `DisabledDocStore` that would return errors for all method calls, and make that the default for the generic. WDYT? Or other ideas on how to deal with "we need a generic for this but want a disabled default"?